### PR TITLE
DSD-897: Updating Required text in the Label and Fieldset components

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-897](https://jira.nypl.org/browse/DSD-897)

## This PR does the following:

- Removes the default "Optional" text that gets displayed by the `Label` and `Fieldset` components. This text was displayed all the way on the right side. Now, only the text "(Required)" is displayed inlined with the label or legend text.
- The "(Required)" text can be hidden with the `showRequiredLabel` prop which is now standardized across all components. This use to be `optReqFlag` and `showOptReqLabel` in various components.

## How has this been tested?

Unit tests and Storybook.

## Accessibility concerns or updates

- This is a visual update and the `required` and `aria-required` attributes in form components have not changed.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr942-vg6ckdg5bzmxqexrcc9uo3f8lklhxulb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624f3c287a1d628954508bee

File: src/components/Label/Label.stories.mdx [hunk 1]
```diff
@@ -55,24 +51,20 @@ import { getCategory } from "../../utils/componentCategories";
...
 <ArgsTable story="Label with Controls" />
 
-## Required/Optional helper text
+## isRequired helper text
 
 An optional helper string can be rendered at the end of the `Label` text through
-the `optReqFlag` prop. Pass in either `"Required"`, `"Optional"`, or `""`. The
-default value is undefined for no helper text to appear.
+the `isRequired` prop. This is useful for visually labeling a field as required
+with the `"(Required"` text next to the label's text. This is set to `false`
```
Comment (bigfishdesign13) [inline src/components/Label/Label.stories.mdx]: Missing closing parenthesis.

**Review by bigfishdesign13 — APPROVED** [review]
Very nice.  One minor typo, but otherwise it all looks good and works well.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr942-vg6ckdg5bzmxqexrcc9uo3f8lklhxulb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624f3c287a1d628954508bee

**Review by oliviawongnyc — APPROVED** [review]
Love this change! Looks good to me.

**Pull request merged**